File: src/main/java/ir/reservs/reservs/ui/splash/SplashPresenter.java
package ir.reservs.reservs.ui.splash;

import android.util.Log;

import java.util.concurrent.TimeUnit;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;
import ir.reservs.reservs.data.DataManager;

public class SplashPresenter implements SplashContract.Presenter {

    private DataManager dataManager;
    private Disposable disposable;
    private SplashContract.View view;

    @Inject
    public SplashPresenter(DataManager dataManager, CompositeDisposable disposable) {
        this.dataManager = dataManager;
        this.disposable = disposable;
    }


    private void decideNextActivity() {
        Log.e("SplashPresenter","decideNextActivity");
        disposable = Observable.timer(1500, TimeUnit.MILLISECONDS).subscribe(aLong -> {
            if (dataManager.getAccessToken() != null) {
                view.openMainActivity();
            } else {
                view.openLoginActivity();
            }
        });
    }

    @Override
    public void onAttach(SplashContract.View view) {
        this.view = view;
        Log.e("SplashPresenter","onAttach");
        decideNextActivity();
    }

    @Override
    public void onDetach() {
        view = null;
        if (!disposable.isDisposed())
            disposable.dispose();
    }
}

File: src/main/java/ir/reservs/reservs/ui/splash/SplashPresenter.kt
package ir.reservs.reservs.ui.splash

import android.util.Log
import io.reactivex.Observable
import io.reactivex.disposables.CompositeDisposable
import ir.reservs.reservs.data.DataManager
import java.util.concurrent.TimeUnit


class SplashPresenter(private val dataManager: DataManager, private val compositeDisposable: CompositeDisposable) : SplashContract.Presenter {
    private var view: SplashContract.View? = null

    private fun decideNextActivity() {
        val disposable = Observable.timer(1500, TimeUnit.MILLISECONDS).subscribe { _ ->
            if (dataManager.getAccessToken() != "-1") {
                view!!.openMainActivity()
            } else {
                view!!.openLoginActivity()
            }
        }
        compositeDisposable.add(disposable)
    }

    override fun onAttach(view: SplashContract.View) {
        this.view = view
        decideNextActivity()
    }


    override fun onDetach() {
        view = null
        if (!compositeDisposable.isDisposed)
            compositeDisposable.dispose()
    }
}
